### PR TITLE
Add Baler module install and config instructions

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request

Added documentation to install and configure the Magento Baler extension to bundle javascript content during the Magento Cloud build process.

## Affected DevDocs pages

https://devdocs.magento.com/cloud/deploy/static-content-deployment.html

## Links to Magento source code

https://github.com/magento/ece-tools/pull/660

whatsnew
Added documentation to install and configure the Magento Baler extension to bundle javascript content during the Magento Cloud build process.

### CODE REVIEW

File: src/cloud/deploy/static-content-deployment.md [hunk 1]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+   >  `.magento.env.yaml`
+
+   ```yaml
+   stage:
+      build:
+         SCD_USE_BALER:
+            - "index.php"
+   ```
+
+1. Update the .magento.app.yaml `hooks` configuration with the Node Version Manager (nvm) configuration that Baler requires to generate the Javascript bundle:
```
Comment (hguthrie) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
1. Update the .magento.app.yaml `hooks` configuration with the Node Version Manager (nvm) configuration that Baler requires to generate the Javascript bundle.
```

File: src/cloud/deploy/static-content-deployment.md [hunk 2]
```diff
@@ -8,7 +8,83 @@ functional_areas:
 
 Static content deployment (SCD) has a significant impact on the store deployment process that depends on how much content to generate—such as images, scripts, CSS, videos, themes, locales, and web pages—and when to generate the content. For example, the default strategy generates static content during the [deploy phase]({{ site.baseurl }}/cloud/deploy/cloud-deployment-process.html#-deploy-phase) when the site is in maintenance mode; however, this deployment strategy takes time to write the content directly to the mounted `pub/static` directory. You have several options or strategies to help you improve the deployment time depending on your needs.
 
-## Minify content
+## Optimize javascript and css content builds
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
## Optimize JavaScript and HTML content
```

File: src/cloud/deploy/static-content-deployment.md [hunk 3]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+You can use bundling and minification to build optimized javascript and html content for static content deployment.
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
You can use bundling and minification to build optimized JavaScript and HTML content during static content deployment.
```

File: src/cloud/deploy/static-content-deployment.md [hunk 4]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+You must install and configure your project to use Baler.
+
+{:.procedure}
+To install and configure Baler:
+
+1. Use the following Baler extension information to [install the the Baler extension]({{ site.baseurl }}/cloud/howtos/install-components.html#install) in a {{site.var.data.ece}} development branch.
+
+   ```text
+   module name: fastly/magento2
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
   module name: magento/module-baler
```

File: src/cloud/deploy/static-content-deployment.md [hunk 5]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+   > `config.php`
+
+   ```php
+       'js' => [
+         'merge_files' => '0',
+         'minify_files' => '0',
+         'enable_js_bundling' => '0',
+         'enable_baler_js_bundling' => '1',
+   ```
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]: There should probably be a bit more context for this (also we typically use 4 spaces to indent):

```php
'system' => [
    'default' => [
        'dev' => [
            'js' => [
                'merge_files' => '0',
                'minify_files' => '0',
                'enable_js_bundling' => '0',
                'enable_baler_js_bundling' => '1',
            ],
        ],
    ],
],
```

File: src/cloud/deploy/static-content-deployment.md [hunk 6]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+   {:.bs-callout-tip}
+   If you do not have a copy of the `config.php` file in your development branch, see [Recommended procedure to manage your settings]({{site.baseurl}}/cloud/live/sens-data-over.html#cloud-config-specific-recomm) to learn how to generate and use this file to manage Magento store configuration for Cloud projects.
+
+1. Update your `.magento.env.yaml` environment configuration file to enable the SCD_USE_BALER option during the build process:
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
1. Update your `.magento.env.yaml` environment configuration file to enable the `SCD_USE_BALER` option during the build process:
```

File: src/cloud/deploy/static-content-deployment.md [hunk 7]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+   >  `.magento.env.yaml`
+
+   ```yaml
+   stage:
+      build:
+         SCD_USE_BALER:
+            - "index.php"
+   ```
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```yaml
stage:
    build:
        SCD_USE_BALER: true
```

File: src/cloud/deploy/static-content-deployment.md [hunk 8]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+1. Update the .magento.app.yaml `hooks` configuration with the Node Version Manager (nvm) configuration that Baler requires to generate the Javascript bundle.
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
1. Update the `.magento.app.yaml` `hooks` configuration to install Node Version Manager (nvm) and the required version of Node for Baler.
```

File: src/cloud/deploy/static-content-deployment.md [hunk 9]
```diff
@@ -8,7 +8,83 @@ functional_areas:
...
+
+   ```yaml
+   hooks:
+      build: |
+         unset NPM_CONFIG_PREFIX
+         curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | dash
+         export NVM_DIR="$HOME/.nvm"
+         [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+         nvm install --lts=dubnium
+   ```
```
Comment (bbatsche) [inline src/cloud/deploy/static-content-deployment.md]: Extra context here is helpful so devs know they need to install NVM & Node _before_ running ECE Tools:
```yaml
hooks:
    build: |
        set -e
        
        unset NPM_CONFIG_PREFIX
        curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | dash
        export NVM_DIR="$HOME/.nvm"
        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
        nvm install --lts=dubnium
        
        php ./vendor/bin/ece-tools build:generate
        php ./vendor/bin/ece-tools build:transfer

```

File: src/cloud/deploy/static-content-deployment.md (hunk 7 above)
Comment (meker12) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
         SCD_USE_BALER:true
   ```
```

File: src/cloud/deploy/static-content-deployment.md (hunk 9 above)
Comment (meker12) [inline src/cloud/deploy/static-content-deployment.md]:
```suggestion
   ```yaml
   hooks:
       build: |
           set -e
        
           unset NPM_CONFIG_PREFIX
           curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | dash
           export NVM_DIR="$HOME/.nvm"
           [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
           nvm install --lts=dubnium
           
           php ./vendor/bin/ece-tools build:generate
           php ./vendor/bin/ece-tools build:transfer
   
   ```
```

**Pull request merged**